lib/workers/global/index.ts

```diff
@@ -104,32 +105,47 @@ export async function resolveGlobalExtends(
 }
 
 export async function start(): Promise<number> {
+  const tracer = getTracer();
   let config: AllConfig;
   try {
-    // read global config from file, env and cli args
-    config = await getGlobalConfig();
-    if (config?.globalExtends) {
-      // resolve global presets immediately
-      config = mergeChildConfig(
-        config,
-        await resolveGlobalExtends(config.globalExtends)
-      );
-    }
-    // initialize all submodules
-    config = await globalInitialize(config);
+    await tracer.startActiveSpan('load config', async (span) => {
+      // read global config from file, env and cli args
+      config = await getGlobalConfig();
+      if (config?.globalExtends) {
+        // resolve global presets immediately
+        config = mergeChildConfig(
+          config,
+          await resolveGlobalExtends(config.globalExtends)
+        );
+      }
+      // initialize all submodules
+      config = await globalInitialize(config);
 
-    // Set platform and endpoint in case local presets are used
-    GlobalConfig.set({ platform: config.platform, endpoint: config.endpoint });
+      // Set platform and endpoint in case local presets are used
+      GlobalConfig.set({
+        platform: config.platform,
+        endpoint: config.endpoint,
+      });
 
-    await validatePresets(config);
+      await validatePresets(config);
 
-    checkEnv();
+      checkEnv();
 
-    // validate secrets. Will throw and abort if invalid
-    validateConfigSecrets(config);
+      // validate secrets. Will throw and abort if invalid
+      validateConfigSecrets(config);
+
+      span.end();
+    });
 
     // autodiscover repositories (needs to come after platform initialization)
-    config = await autodiscoverRepositories(config);
+    config = await tracer.startActiveSpan(
+      'discover repositories',
+      async (span) => {
+        await autodiscoverRepositories(config);
+        span.end();
+        return config;
```

Comment:
```suggestion
        config = await autodiscoverRepositories(config);
        span.end();
        return config;
```

Shouldn't it be like this to preserve the existing logic?

Reply:
Would `tracer.startActiveSpan` auto end the span when there is an exception thrown?

Reply:
No, but I have added an instrument function to handle this and replaced all usages of `startActiveSpan`